### DISCUSSION TITLE
MetroLog not closing file handle - bug or intentional?

### DISCUSSION THREAD

**Discussion by COMH-AIQ** [roubachof/MetroLog#21] [General]
I'm using the latest release of MetroLog.Maui in a .NET 7.0 application targeting Windows [unpackaged]. I noticed that MetroLog does not seem to close the file handle after writing a log entry. If I open my app and let it sit for a while, I am unable to delete the log file as long as the application is running. From using other logging libraries, this is unexpected. I'm used to a log entry to write, flush, close the file. 

![MetroLoggingFileHold](https://user-images.githubusercontent.com/94493438/230124006-69a4511a-63f6-439a-b54a-d0fd3763a144.jpg)

**Comment by roubachof** [comment]
In fact this is expected for mobile targets like iOS or Android, cause it gives way better performance. 
But we should close the handle after each write on desktop platform.
